client/client.go

```diff
@@ -394,9 +394,35 @@ type RemoteContext struct {
 	// preference.
 	Platforms []string
 
-	// MaxConcurrentDownloads is the max concurrent content downloads for each pull.
+	// MaxConcurrentDownloads restricts the total number of concurrent downloads
+	// across all layers during an image pull operation. This helps control the
+	// overall network bandwidth usage.
 	MaxConcurrentDownloads int
 
+	// MaxConcurrentDownloadOperations limits the number of parallel operations
+	// during an image pull to prevent system resource exhaustion.
+	//
+	// These operations include:
+	// - Downloading a layer
+	// - Unpacking a layer into the snapshotter
+	// - Moving an unpacked layer to its final location
+	MaxConcurrentDownloadOperations int
```

Comment:
@azr can we actually remove this config? I feel this one is a bit confusing and not sure how to config an appropriate value. 

Previously we only have `MaxConcurrentDownloads` and IIUC we  didn't limit other ops like unpacking/moving? (actually can you explain difference between op Unpacking and op 3 Moving).

Reply:
Trying, that's actually a think. I wanted to propose, maybe also the other configuration settings ( and just bool activate multipart_fetch ? ).
When deploying this containerd, it's a bit tedious to optimize a per node config config, and maybe we could infer that setting from cpu count, or something similar.

Reply:
> actually can you explain difference between op Unpacking and op 3 Moving

When I was debugging things I saw this operation in the `handler` loop, now that I think about it, it shouldn't exist, should it ? removing this anyways, will try to debug and repro to explain better if time.

Reply:
> actually can you explain difference between op Unpacking and op 3 Moving

There is no moving. If I got things correctly, we:
* open a bytestream to dl a file 
( sometimes that file is a manifest which will add more file to dl in step zero ) 
* put it in a tar
* open a bytestream to that tar
* apply a diff from it to a read-only fs

each individual step could be limited here.

Reply:
Also re, about the preferred settings, when prodding this, I have been using the following setting everywhere:

```
    max_concurrent_downloads=32
    max_concurrent_downloads_per_layer=32
    concurrent_download_chunk_size=8388608
```

Instead of trying to guess a setting per machine. This setting proves to be bang on for almost every occasion, and we always end up being throttled by unpacking.
Some image downloads went from 2min to 25s/30s on big machines.

Reply:
Please lmk if I can improve anything ! 😊